lib/spack/spack/build_environment.py

```diff
@@ -779,19 +779,19 @@ def make_stack(tb, stack=None):
     # point out the location in the install method where we failed.
     lines = []
     lines.append("%s:%d, in %s:" % (
-        inspect.getfile(frame.f_code), frame.f_lineno, frame.f_code.co_name
+        inspect.getfile(frame.f_code), frame.f_lineno - 1, frame.f_code.co_name
```

Comment:
could you add a comment explaining why the -1 is needed, otherwise this is magic for the next reader

Reply:
I agree with @healther